frontend/packages/ui/src/components/menu-handler/menu-handler.tsx

```diff
@@ -39,9 +39,15 @@ export function MenuHandler({ customItems, className }: Props) {
   const openMenu = () => setIsMenuOpen(true);
   const closeMenu = () => setIsMenuOpen(false);
 
-  useClickOutside(() => {
-    closeMenu();
-  }, menuRef);
+  const modalRootRef = useRootModalRef();
```

Comment:
Let's leave a comment with explanation of why we need modal root here, since it seems a little unobvious. I guess eventually we will come back to this problem.